Client/game_sa/CEntitySA.cpp

```diff
@@ -628,6 +650,33 @@ bool CEntitySA::SetBoneRotation(eBone boneId, float yaw, float pitch, float roll
     return false;
 }
 
+bool CEntitySA::SetBoneRotationQuat(eBone boneId, float x, float y, float z, float w)
+{
+    RpClump* clump = GetRpClump();
+    if (clump)
+    {
+        // updating the bone frame orientation will also update its children
+        // This rotation is only applied when UpdateElementRpHAnim is called
+        CAnimBlendClumpDataSAInterface* clumpDataInterface = *pGame->GetClumpData(clump);
+        AnimBlendFrameData*             frameData = clumpDataInterface->GetFrameDataByNodeId(boneId);
+        if (frameData)
+        {
+            RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
+            boneOrientation->imag.x = x;
+            boneOrientation->imag.y = y;
+            boneOrientation->imag.z = z;
+            boneOrientation->real = w;
+            CEntitySAInterface* theInterface = GetInterface();
+            if (theInterface)
+            {
+                theInterface->bDontUpdateHierarchy = false;
+            }
+            return true;
+        }
+    }
+    return false;
```

Comment:
Still early return here and above.

Reply:
this isn't necessary, I'm not here refactoring multi theft auto repo but adding new functions. please stop making pointless comments blocking pull requests from being merged.

Reply:
> this isn't necessary, I'm not here refactoring multi theft auto repo but adding new functions. please stop making pointless comments blocking pull requests from being merged.

When you modify code, you have to follow the guidelines.
You created functions so its not refactoring.